src/main/java/pixelperfect/Spaceship.java

```diff
@@ -15,6 +16,7 @@
 public class Spaceship {
 
 	private static final Spaceship ship = new Spaceship();
+	private RouteGenerator routeGenerator;
```

Comment:
Of course, that is why the Route uses the RouteGenerator. Giving the RouteGenerator away to another class creates the risk of the Route becoming a data class.